.github/workflows/container-image.yml

```diff
@@ -18,3 +18,20 @@ jobs:
       run: make build
     - name: Test the image
       run: make test
+
+  push:
+
+    # verify it passes build and test first
+    needs: build
+
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
```

Comment:
Does this include merges post-pull-request?

Reply:
yes, direct push and pushes/merges of a PR (is my understanding)

Reply:
of course we could also create a separate workflow yaml just for the pushing, if that's more obvious. but my understanding is that this event_name directly relates to the `on:` section above

Reply:
It just wasn't too obvious to me w/o digging through the github actions documentation. We can see if it works in a practical manner soon. :-)